plugin/kubernetes/transfer.go

```diff
@@ -0,0 +1,216 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"math"
+	"net"
+	"strings"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/etcd/msg"
+	"github.com/coredns/coredns/plugin/transfer"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+	api "k8s.io/api/core/v1"
+)
+
+// Transfer implements transfer.Transferer
+func (k *Kubernetes) Transfer(zone string, serial uint32) (<-chan []dns.RR, error) {
+	ch := make(chan []dns.RR, 2)
```

Comment:
make the channel later, so you don't need close it if the zone isn't found

Reply:
I think in my PR I can get away with unbuffered and a go routine here. https://github.com/coredns/coredns/pull/3972/files#diff-f1591d056edea707afdeae51b8dc8b8fR25 onward